File: internal/src/main/java/com/striveonger/music/fairy/sources/api/Music.java
package com.striveonger.music.fairy.sources.api;

import java.util.List;

/**
 * @author Mr.Lee
 * @description:
 * @date 2024-07-02 20:01
 */
public interface Music {

    String type();

    /**
     * 转换成可播放音频的对象
     *
     * @param url 可播放的流媒体地址
     * @return
     */
    <T extends Play> List<T> convert(String url);

    <T extends Play> byte[] play(T play);

    /**
     * 搜索得到列表
     */
    List<SearchItem> search(String keyword, int page);

    default List<SearchItem> search(String keyword) {
        return search(keyword, 1);
    }

}

File: website/src/apis/Music.ts
import { Result, http } from "./index";
import { SearchItem, Play, BilibiliPlay } from "../types";

export async function search(keyword: string, page: number): Promise<SearchItem[]> {
    const result = await http.get<Result<SearchItem>>("/api/v1/fairy/music/search", { params: { keyword, page } });
    return result.data.data as SearchItem[];
}

export async function playlist(url: string): Promise<Play[]> {
    // http.get<Result<Play>>("/api/v1/fairy/music/playlist", { params: { url } }).then(x).catch(y);
    const response = await http.get<Result<Play>>("/api/v1/fairy/music/playlist", { params: { url } });
    return response.data.data as Play[];
}